File: src/Image4ioSDK/IClient.vb
Imports Image4ioSDK
Imports Image4ioSDK.JSON

Public Interface IClient


    ReadOnly Property Transformation(ImageMetadata As JSON_ImageMetadata) As ITransformation


    Function ImageMetadata(ImagePath As String) As Task(Of JSON.JSON_ImageMetadata)
    Function Upload(FileToUpload As Object, UploadType As utilitiez.UploadTypes, DestinationFolderPath As String, FileName As String, UseFilename As Boolean, OverwriteIfExists As Boolean, Optional ReportCls As IProgress(Of ReportStatus) = Nothing, Optional token As Threading.CancellationToken = Nothing) As Task(Of JSON_Uploadedfile)
    Function UploadMultiable(FilesToUpload As List(Of OClient.MultiableUpload), UploadType As utilitiez.UploadTypes, DestinationFolderPath As String, UseFilename As Boolean, OverwriteIfExists As Boolean, Optional ReportCls As IProgress(Of ReportStatus) = Nothing, Optional token As Threading.CancellationToken = Nothing) As Task(Of JSON_MultiUpload)
    Function UploadRemotely(ImgUrl As String, DestinationFolderPath As String) As Task(Of JSON_Uploadedfile)

    Function ImageCopy(ImagePath As String, DestinationFolderPath As String) As Task(Of JSON_Uploadedfile)
    Function ImageMove(ImagePath As String, DestinationFolderPath As String) As Task(Of JSON_Uploadedfile)
    Function ImageDelete(ImagePath As String) As Task(Of JSON_Uploadedfile)

    Function FolderDelete(FolderPath As String) As Task(Of JSON_Uploadedfile)
    Function FolderCreate(FolderName As String, DestinationFolderPath As String) As Task(Of JSON_Uploadedfile)
    Function FolderList(FolderPath As String) As Task(Of JSON_FolderList)

End Interface

File: Image4ioSDK/Image4ioSDK/IClient.cs
using Image4ioSDK.JSON;
using System;
using System.Collections.Generic;
using System.Threading;
using System.Threading.Tasks;

namespace Image4ioSDK
{
    public interface IClient
    {
        //public ITransformation Transformation(JSON_ImageMetadata ImageMetadata) { get; }
        ITransformation Transformation(JSON_ImageMetadata ImageMetadata);


        Task<JSON.JSON_ImageMetadata> ImageMetadata(string ImagePath);
        Task<JSON_Uploadedfile> Upload(object FileToUpload, Utilitiez.UploadTypes UploadType, string DestinationFolderPath, string FileName, bool UseFilename, bool OverwriteIfExists, IProgress<ReportStatus> ReportCls = null, System.Threading.CancellationToken token = default(CancellationToken));
        Task<JSON_MultiUpload> UploadMultiable(List<OClient.MultiableUpload> FilesToUpload, Utilitiez.UploadTypes UploadType, string DestinationFolderPath, bool UseFilename, bool OverwriteIfExists, IProgress<ReportStatus> ReportCls = null, System.Threading.CancellationToken token = default(CancellationToken));
        Task<JSON_Uploadedfile> UploadRemotely(string ImgUrl, string DestinationFolderPath);

        Task<JSON_Uploadedfile> ImageCopy(string ImagePath, string DestinationFolderPath);
        Task<JSON_Uploadedfile> ImageMove(string ImagePath, string DestinationFolderPath);
        Task<JSON_Uploadedfile> ImageDelete(string ImagePath);

        Task<JSON_Uploadedfile> FolderDelete(string FolderPath);
        Task<JSON_Uploadedfile> FolderCreate(string FolderName, string DestinationFolderPath);
        Task<JSON_FolderList> FolderList(string FolderPath);
        Task Download(string ImageUrl, string ImageSaveDir, string FileName, IProgress<ReportStatus> ReportCls = null, CancellationToken token = default(CancellationToken));


    }
}
